---
title: "Instituto de Biomedicina de Sevilla"
description: "Gene Expression & Bioinformatics"
weight: 9
---

<script src="/rmarkdown-libs/header-attrs/header-attrs.js"></script>


<style>
    p {text-align: justify;}
</style>
<div id="gene-expression-lab-instituto-de-biomedicina-de-sevilla" class="section level2">
<h2><a href="https://www.ibis-sevilla.es/investigacion/oncohematologia-y-genetica/expresion-genica/chavez-de-diego-sebastian.aspx" target="_blank">Gene expression lab</a>, <a href="https://www.ibis-sevilla.es/"  target="_blank"><i>Instituto de Biomedicina de Sevilla</i></a></h2>
<p style="font-size:15px;color:gray">
<em style='color:black;'><em>September 2017 - October 2018</em></em><br>
Master’s thesis student and research assistant
</p>
<hr />
<p>In the context of the master’s thesis (<em>MSc in Molecular Genetics &amp; Biotechnology</em>), under the supervision of Dr. Sebastián Chávez de Diego and Dr. Gabriel Gutierrez Pozo and within Dr. Victoria Begley’s PhD project, I started studying Xrn1, a protein involved in the degradation of mRNA, and it’s role in the nucleosome localization and transcription in <em>Saccharomyces cerevisiae</em>. Using MNase-seq and following a DANPOS-based analysis, we found that the Xrn1 knockout mutant strains had higher nucleosome occupancy around TSS, which we related with a lower transcription initiation compared to WT strains, whereas in the gene body, the nucleosome occupancy was diminished as a consequence of RNA pol II backtracking. Results were validated using MNase-qPCR of two selected genes in Xrn1 conditional mutants.</p>
<hr />
<ins>
My contributions to this research were:
</ins>
<ul>
<li>Bioinformatic analysis of the MNase-seq data. <br></li>
<li>MNase-qPCR validation, including the yeast culture, the MNase-driven DNA digestion, the design of the primers and the following qPCRs.</li>
</ul>
<hr />
<ins>
Master’s thesis:
</ins>
<ul>
<li><em>Influence of Xrn1 in nucleosome positioning across </em>Saccharomyces cerevisiae<em> genome and its effects in transcription.</em></li>
</ul>
<hr />
<ins>
Related publications:
</ins>
<ul>
<li>Begley <em>et al</em> (2020). <strong>Xrn1 influence on gene transcription results from the combination of general effects on elongating RNA pol II and gene-specific chromatin configuration</strong>. <em>RNA Biology</em>. <a href="https://doi.org/10.1080/15476286.2020.1845504">doi: 10.1080/15476286.2020.1845504</a>.</li>
</ul>
</div>
