---
title: "Istituto Europeo di Oncologia"
description: "Epigenetics & Bioinformatics"
weight: 7
---

<script src="/rmarkdown-libs/header-attrs/header-attrs.js"></script>


<div id="pasinis-lab-istituto-europeo-di-oncologia" class="section level2">
<h2>Pasiniâ€™s lab, <i>Istituto Europeo di Oncologia</i></h2>
<p style="font-size:15px;color:gray">
<em style='color:black;'><em>July 2019 - Present</em></em><br>
Bioinformatician and PhD student
</p>
<hr />
<p align="justify">
Insert research here
</p>
<!--
<p align='justify'>
With the goal of completing a PhD thesis, I have been involved in the analysis of transcriptomics and epigenomics data to study the effects of mutations on the Wnt pathway that stabilize &beta;catenin, constitutively overactivate the pathway and cause mouse organoids and stem cells from small intestine acquire adenoma-like phenotype and characteristics.
</p>
-->
<hr />
<p><span style="text-decoration:underline">My contribution to this research was:</span></p>
<p align="justify">
<ul>
<li>Comparison of different ATACseq peak callers. <br></li>
<li>Comparison of different sequencing depths in RNAseq. <br></li>
<li>Development of an ATACseq pipeline to run using Snakemake. The code can be visualized <a href="https://github.com/amitjavilaventura/ATACseq-snakemake">here</a> (still ongoing). <br></li>
<li>Development of a ShinyApp to allow wetlab members to visualize the omics produced in the lab in an easy way. The code can be visualized <a href="https://github.com/amitjavilaventura/DPomics">here</a>. <br></li>
<li>Analysis of transcriptomics (RNAseq), cistromics (ChIPseq) and DNA accessibility data (ATACseq). <br></li>
</ul>
<!--
* Analysis of the CRC-related ignorome.
-->
<!--
---
<span style="text-decoration:underline">Related publications:</span>
-->
<!--
---
<span style="text-decoration:underline">Collaborations:</span>
-->
</p>
</div>
